index.bs

```diff
@@ -1099,18 +1099,25 @@ optional |expires|,
 |sameSite|,
 run the following steps:
 
+1. If |name|, |value|, |domain|, or |path| contain U+003B (`;`), any [=C0 control=] character, or U+007F, then return failure.
 1. If |name|'s [=string/length=] is 0 and |value| contains U+003D (`=`), then return failure.
 1. If |name|'s [=string/length=] is 0 and |value|'s [=string/length=] is 0, then return failure.
+1. Determine whether |name| and |value| together exceed cookie size limits using these steps:
+    1. [=UTF-8 encode=] |name| and then compute its [=byte sequence=] [=byte sequence/length=].
+    1. [=UTF-8 encode=] |value| and then compute its [=byte sequence=] [=byte sequence/length=].
+    1. If the sum of the two computed lengths is greater than 4096 [=byte|bytes=], then return failure.
 1. Let |host| be |url|'s [=url/host=]
 1. Let |attributes| be a new [=/list=].
 1. If |domain| is not null, then run these steps:
+    1. If |domain|'s [=UTF-8 encode|UTF-8-encoded=] [=byte sequence=] [=byte sequence/length=] is greater than 1024 [=byte|bytes=], then return failure.
```

Comment:
Maybe introduce _domainEncoded_ here too, and break this into two steps?

Also, maybe move this to be the last test/possible failure before appending?